VogtObserver/Program.cs

```diff
@@ -8,33 +8,26 @@ class Program
         static void Main(string[] args)
         {
             RandomNumberGenerator generator = new RandomNumberGenerator();
-            Date dateTime = new Date();
-            Sum sum = new Sum();
+            Date dateTime = new Date(generator);
+            Sum sum = new Sum(generator);
 
             Run(generator, dateTime, sum);
         }
 
         public static void Run(RandomNumberGenerator generator, Date dateTime, Sum sum)
         {
-            generator.Add(dateTime);
-            generator.Add(sum);
 
-            for (int i = 1; i <=20 ; i++)
-            {
-                try
-                {
-                    Console.Clear();
+            dateTime = new Date(generator);
+            sum = new Sum(generator);
 
-                    generator.Notify();
+            for (int i = 0; i < 20; i++)
+            {
+                Console.Clear();
+                generator.GetRandomInt();
```

Comment:
This looks good here! I'd still rename GetRandomInt to `ChangeToRandomNumber`. That more accurate describes the behavior. And it makes more sense in terms of observers vs observable. That random number is observable, so any changes are watched.